t/scripts/event-trace.lua

```diff
@@ -84,5 +78,17 @@ local mh, err = f:msghandler {
 	end
     end
 }
+if err then
+   io.stderr:write("Error in registering exit event watcher\n")
+   os.exit(1)
+end
+if f:subscribe (s) < 0 then
```

Comment:
I don't think the Lua `subscribe` method returns `< 0` on error. Lua idiom is to return `nil, error` on error, so the usual way to check for error is to do
```lua
local rc, err = f:subscribe (s)
if not rc then
    io.stderr:write ("subscribe: "..err)
end
```

another way to perform this check is to use the Lua `assert()` function
```lua
assert (f:subscribe (s))
```

which errors out with the second return value if the first return value is `nil` or `false`.

Reply:
Thanks! Good to know.

Updated.